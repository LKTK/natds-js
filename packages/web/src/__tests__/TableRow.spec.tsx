import React from 'react';
import * as renderer from 'react-test-renderer';

import {
    Table,
    TableCell,
    TableContainer,
    TableHead,
    TableRow,
} from '..';

describe('TableRow component', () => {
    describe('rendering variants', () => {

        test('should match to snapshot - standard TableRow', () => {
            const component = renderer.create(
                <TableContainer>
                    <Table dividers={true}>
                        <TableHead>
                            <TableRow selected>
                                <TableCell>Test table cell text</TableCell>
                                <TableCell>Test table cell text</TableCell>
                                <TableCell>Test table cell text</TableCell>
                            </TableRow>
                        </TableHead>
                    </Table>
                </TableContainer>
            );
            expect(component.toJSON()).toMatchSnapshot(
                'TableComponent with dividers'
            );
        });

        test('should match to snapshot - TableRow with selected on', () => {
            const component = renderer.create(
                <TableContainer>
                    <Table dividers={true}>
                        <TableHead>
                            <TableRow selected={true}>
                                <TableCell>Test table cell text</TableCell>
                                <TableCell>Test table cell text</TableCell>
                                <TableCell>Test table cell text</TableCell>
                            </TableRow>
                        </TableHead>
                    </Table>
                </TableContainer>
            );
            expect(component.toJSON()).toMatchSnapshot(
                'TableComponent with dividers'
            );
        });
    });
});