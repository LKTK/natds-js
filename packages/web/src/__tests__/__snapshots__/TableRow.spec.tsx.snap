// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow component rendering variants should match to snapshot - TableRow with selected on: TableComponent with dividers 1`] = `
<div
  className="MuiTableContainer-root"
>
  <table
    className="MuiTable-root makeStyles-root-3"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head Mui-selected makeStyles-selected-2"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Test table cell text
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Test table cell text
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Test table cell text
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`TableRow component rendering variants should match to snapshot - standard TableRow: TableComponent with dividers 1`] = `
<div
  className="MuiTableContainer-root"
>
  <table
    className="MuiTable-root makeStyles-root-1"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head Mui-selected makeStyles-selected-2"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Test table cell text
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Test table cell text
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Test table cell text
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
